api.planx.uk/modules/gis/service/digitalLand.ts

```diff
@@ -71,15 +71,30 @@ async function go(
   geom: string,
   extras: Record<string, string>,
 ): Promise<GISResponse> {
-  // generate list of digital land datasets we should query based on 'active' planx schema variables
+  // generate list of digital land datasets we should query and their associated passport values
   const activeDatasets: string[] = [];
-  Object.keys(baseSchema).forEach((key) => {
-    if (baseSchema[key]["active"]) {
-      baseSchema[key]["digital-land-datasets"]?.forEach((dataset: string) => {
-        activeDatasets.push(dataset);
-      });
-    }
-  });
+  let activeDataValues: string[] = [];
+  if (extras?.vals) {
+    // if the editor selected constraints, prioritise their selection
+    activeDataValues = extras.vals.split(",");
+    Object.keys(baseSchema).forEach((key) => {
+      if (activeDataValues.includes(key)) {
+        baseSchema[key]["digital-land-datasets"]?.forEach((dataset: string) => {
+          activeDatasets.push(dataset);
+        });
+      }
+    });
+  } else {
+    // else default to the internally maintained list of all "active" datasets
+    Object.keys(baseSchema).forEach((key) => {
+      if (baseSchema[key]["active"]) {
+        activeDataValues.push(key);
+        baseSchema[key]["digital-land-datasets"]?.forEach((dataset: string) => {
+          activeDatasets.push(dataset);
+        });
+      }
+    });
+  }
```

Comment:
All _new_ Planning Constraint requests will include `vals` param, but we also want to support existing "legacy" GIS requests that have been shared via ODP Schema application payloads (eg BOPS uses to "re-fetch" constraints). 

Therefore, both of these will work: 
- (with `vals` only fetches selected datasets) http://localhost:7002/gis/opensystemslab?geom=MULTIPOLYGON+%28%28%28-0.076303+51.485923%2C+-0.076307+51.48591%2C+-0.075586+51.485849%2C+-0.075574+51.485849%2C+-0.075569+51.485903%2C+-0.076293+51.485963%2C+-0.076303+51.485923%29%29%29&vals=article4
- (without `vals` defaults to all "active" datasets as before) http://localhost:7002/gis/opensystemslab?geom=MULTIPOLYGON+%28%28%28-0.076303+51.485923%2C+-0.076307+51.48591%2C+-0.075586+51.485849%2C+-0.075574+51.485849%2C+-0.075569+51.485903%2C+-0.076293+51.485963%2C+-0.076303+51.485923%29%29%29

Reply:
I guess this actually isn't _totally_ stable just now in the same way that the Planning Data `sourceRequest` URL is. If in future there are new datasets added, the old URLs will pick up the full current list and not the one used at the time.

Bit of an edge case, but something to flag 👍